src/core/qgsogcutils.cpp

```diff
@@ -3370,8 +3370,7 @@ QgsExpressionNode *QgsOgcUtilsExpressionFromFilter::nodeLiteralFromOgcFilter( co
           if ( fieldIndex != -1 )
           {
             QgsField field = mLayer->fields().field( propertyNameElement.firstChild().nodeValue() );
-            field.convertCompatible( value );
-            converted = true;
+            converted = field.convertCompatible( value );
```

Comment:
I think it's not needed anymore thanks to https://github.com/qgis/QGIS/pull/30290, no?

Reply:
I don't know. I was thinking that convertCompatible returns true if the value has been updated so converted can be false.

Reply:
Can you provide a test case for this modification? I'll try to take a look.

Reply:
I do not have a test case. I can remove these changes. I was thinking, it was better to do like that but I am not sure its is.

Reply:
Do I remove this commit 27d6bc707dacc3cce3741e01463b38a08670b457 ?

Reply:
I've to admit that without unit tests, I'd be more in favor of removing this commit yes.

Reply:
done